package linklayer;

message nodeStats {
  required string address = 1;
  required int32 packets = 2;
}

message macStatistics {
  optional int32 txQueueSize = 1;
  optional int32 txQueueCapacity = 2;
  optional int32 txPerPacket = 3;
  optional int32 txRate = 4;
  optional int32 minCw = 5;
  optional int32 maxCw = 6;
  optional int32 curCw = 7;
  optional float rtt = 8;
  optional float pktSize = 9;
  repeated nodeStats txStats = 10;
  repeated nodeStats rxStats = 11;
}
